activity.go

```diff
@@ -57,8 +62,27 @@ type (
 // Serialization of all primitive types, structures is supported ... except channels, functions, variadic, unsafe pointer.
 // This method calls panic if activityFunc doesn't comply with the expected format.
 func RegisterActivity(activityFunc interface{}) {
```

Comment:
The intention was not backwards compatibility. In a lot of cases users will not provide a name, having to add an empty struct is not a great user experience. So they can simple call RegisterActivity(fn) if they dont have an external name.